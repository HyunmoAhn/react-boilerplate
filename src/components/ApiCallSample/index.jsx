import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	data: PropTypes.array,
	onFetchApiCall: PropTypes.func,
};

const defaultProps = {
	data: [],
	onFetchApiCall() {},
};

class ApiCallSample extends React.Component {
	render() {
		const { data, onFetchApiCall } = this.props;

		return (
			<div>
				<button
					type="button"
					onClick={onFetchApiCall}
				>
					ApiCall!
				</button>
				{data.map((v, index) => <div key={index}>{v}</div>)}
			</div>

		);
	}
}

ApiCallSample.propTypes = propTypes;
ApiCallSample.defaultProps = defaultProps;

export default ApiCallSample;