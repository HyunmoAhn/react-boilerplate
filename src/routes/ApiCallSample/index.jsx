import ApiCallSampleContainer from './containers/ApiCallSampleContainer';

export default () => ({
  path: 'api-call',
  component: ApiCallSampleContainer,
});
