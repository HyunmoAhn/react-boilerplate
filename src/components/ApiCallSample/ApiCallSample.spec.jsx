import React from 'react';
import { shallow } from 'enzyme';

import ApiCallSample from './';

describe('<ApiCallSample />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<ApiCallSample />);

    expect(wrapper).toMatchSnapshot();
  });
});
