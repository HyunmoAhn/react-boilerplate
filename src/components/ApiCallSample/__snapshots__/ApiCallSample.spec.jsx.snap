// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApiCallSample /> should match to snapshot when render default 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    ApiCall!
  </button>
</div>
`;
